Code/Framework/AzCore/AzCore/Math/LineSegment.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Math/Vector3.h>
+#include <AzCore/RTTI/TypeInfoSimple.h>
+
+namespace O3DE
+{
+    class LineSegment
+    {
+    public:
+        AZ_TYPE_INFO(LineSegment, "{7557da1e-cc20-11ec-9d64-0242ac120002}");
+
+        explicit LineSegment(const AZ::Vector3& start, const AZ::Vector3& end);
+
+        const AZ::Vector3& GetStart() const;
+        const AZ::Vector3& getEnd() const;
+
```

Comment:
Suggestion: Some other methods that might be useful on this class:
* GetLength() - get the length of the segment
* GetDirection() - get the (normalized?) direction of the segment
* GetPoint(float t) - Given a distance t [0, 1] on the segment, get the point on it
* IsPointOnSegment(const AZ::Vector3& point) - Given a point, return whether or not it's on the segment